server/core/terraform/downloader.go

```diff
@@ -0,0 +1,70 @@
+package terraform
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/go-version"
+	install "github.com/hashicorp/hc-install"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/hc-install/src"
+	"github.com/opentofu/tofudl"
+)
+
+//go:generate pegomock generate --package mocks -o mocks/mock_downloader.go Downloader
+
+// Downloader is for downloading terraform versions.
+type Downloader interface {
+	Install(dir string, downloadURL string, v *version.Version) (string, error)
+}
+
+type TofuDownloader struct{}
+
+func (d *TofuDownloader) Install(dir string, _downloadURL string, v *version.Version) (string, error) {
+	// Initialize the downloader:
+	dl, err := tofudl.New()
+	if err != nil {
+		return "", err
+	}
+
+	binary, err := dl.Download(context.Background(), tofudl.DownloadOptVersion(tofudl.Version(v.String())))
+	if err != nil {
+		return "", err
+	}
+
+	// Write out the tofu binary to the disk:
+	file := filepath.Join(dir, "terraform"+v.String())
```

Comment:
writing the tofu binary as "terraform." Is this intentional to avoid refactoring more of the codebase even though we track the name via distribution.go?